### PR TITLE
[INFRA-6584] Added docker files;

### PR DESCRIPTION
**Pull request by alexanderilyin**
Added `Dockerfile` and `docker-compose.yml` and changed directory structure to increase cache hit for docker build cache.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	&& mkdir -p /usr/share/nginx/html/ \
+	&& install -m644 html/index.html /usr/share/nginx/html/ \
+	&& install -m644 html/50x.html /usr/share/nginx/html/ \
+	&& install -m755 objs/nginx-debug /usr/sbin/nginx-debug \
+	&& install -m755 objs/ngx_http_xslt_filter_module-debug.so /usr/lib/nginx/modules/ngx_http_xslt_filter_module-debug.so \
+	&& install -m755 objs/ngx_http_image_filter_module-debug.so /usr/lib/nginx/modules/ngx_http_image_filter_module-debug.so \
+	&& install -m755 objs/ngx_http_geoip_module-debug.so /usr/lib/nginx/modules/ngx_http_geoip_module-debug.so \
+	&& install -m755 objs/ngx_stream_geoip_module-debug.so /usr/lib/nginx/modules/ngx_stream_geoip_module-debug.so \
+	&& ln -s ../../usr/lib/nginx/modules /etc/nginx/modules \
+	&& strip /usr/sbin/nginx* \
```
Comment (ahammond) [inline Dockerfile:112]: Please don't strip binaries unless it is an absolute requirement.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	&& apk add --no-cache tzdata \
+	\
+	# forward request and error logs to docker log collector
+	&& ln -sf /dev/stdout /var/log/nginx/access.log \
+	&& ln -sf /dev/stderr /var/log/nginx/error.log
+
+COPY nginx.conf /etc/nginx/nginx.conf
+COPY nginx.vh.default.conf /etc/nginx/conf.d/default.conf
+
+EXPOSE 80
```
Comment (ahammond) [inline Dockerfile:145]: No 443? Good. SSL should be handled by haproxy. But if we're going to assume haproxy, the next question is if the metrics already exposed by haproxy are sufficient to render pinba metrics un-necessary. WDYT.

**Review by ahammond — APPROVED** [review]
I'd like to see your thoughts about the metrics already provided by haproxy vs. what we'd get from nginx pinba.

File: Dockerfile (hunk 2 above)
Comment (alexanderilyin) [reply Dockerfile:145]: @ahammond could you please provide links to raw data/ visualizations which you are talking about? Pinba still could be useful for example to get metrics for each `location`.

**Pull request closed without merging**